custom_components/isy994/__init__.py

```diff
@@ -421,40 +353,31 @@ def _categorize_programs(hass_isy_data: dict, programs: dict) -> None:
                 hass_isy_data[ISY994_PROGRAMS][platform].append(entity)
 
 
-def _categorize_variables(
-    hass_isy_data: dict, variables, platform_cfg: dict, platform: str
-) -> None:
-    """Categorize the ISY994 Variables."""
-    if platform_cfg is None:
-        return
-    for isy_var in platform_cfg:
-        vid = isy_var.get(CONF_ID)
-        vtype = isy_var.get(CONF_TYPE)
-        vname = ""
-        try:
-            vname = variables[vtype][vid].name
-        except KeyError as err:
-            _LOGGER.error("Error adding ISY Variable %s.%s: %s", vtype, vid, err)
-            continue
-        else:
-            variable = (isy_var, vname, variables[vtype][vid])
-            hass_isy_data[ISY994_VARIABLES][platform].append(variable)
+def _categorize_variables(hass_isy_data: dict, variables, identifier: str) -> None:
+    """Gather the ISY994 Variables to be added as sensors."""
+    try:
+        var_to_add = [
+            (vtype, vname, vid)
+            for (vtype, vname, vid) in variables.children
+            if identifier in vname
+        ]
+    except KeyError as err:
+        _LOGGER.error("Error adding ISY Variables: %s", err)
+    else:
+        for vtype, vname, vid in var_to_add:
+            hass_isy_data[ISY994_VARIABLES].append((vname, variables[vtype][vid]))
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the isy994 component from YAML."""
     isy_config: Optional[ConfigType] = config.get(DOMAIN)
-    hass.data.setdefault(DOMAIN, {})
```

Comment:
Tested and found that this need to stay as startup will fail if there is no yaml config

```
    hass.data.setdefault(DOMAIN, {})
```


```
2020-05-02 19:30:05 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Alexander (192.168.209.32) for isy994
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 216, in async_setup
    hass, self
  File "/config/custom_components/isy994/__init__.py", line 413, in async_setup_entry
    hass.data[DOMAIN][entry.entry_id] = {}
KeyError: 'isy994'
```

Reply:
It's still there, just on the other side of the `if not` check.